receiver/awsebsnvmereceiver/internal/nvme/device_file_attributes.go

```
@@ -0,0 +1,67 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package nvme
+
+import (
+	"errors"
+	"fmt"
+)
+
+type DeviceFileAttributes struct {
+	controller int
+	namespace  int
+	partition  int
+}
+
+func ParseNvmeDeviceFileName(device string) (DeviceFileAttributes, error) {
+	controller := -1
+	namespace := -1
+	partition := -1
+
+	fmt.Sscanf(device, "nvme%dn%dp%d", &controller, &namespace, &partition)
```

can we check if this returns an error? if so then we are unable to parse the device

---

I opted to ignore the error and chose to throw an error if we couldn't at least get the controller ID. The reason is that this function parses the three different device file name patterns.

I could instead just do something like check for nvme%dn%dp%d, then nvme%dn%d, and lastly nvme%d. I guess that'll give a better error message but it seems like a lot of work. In which then I kind of like the implementation I had before.